### DISCUSSION TITLE
DB Architecture

### DISCUSSION THREAD

**Discussion by dmarro89** [dmarro89/dare-db#12] [Ideas]
Starting from @vdmitriyev diagram, here's the discussion about the DB architecture.

I had created an initial draft of an architecture. It is in a ```mermaid``` format. Should we have a different discussion thread on that as well? Here is an architecture:

```mermaid
graph LR
    subgraph connection
        HTTPS(HTTPS)
        SOC(socket)
    end

    subgraph auth
        HTTPS --> RBAC(RBAC)
        SOC --> RBAC(RBAC)
    end
    
    subgraph engine
        ENG --> DAT[(data)]
        RBAC --> ENG(engine)
    end

    subgraph filesystem
        ENG --> CONF(config.yml) 
    end 
    
```

_Originally posted by @vdmitriyev in https://github.com/dmarro89/dare-db/discussions/1#discussioncomment-9545336_

**Comment by vdmitriyev** [comment]
Discussion of data structures for DareDB: https://github.com/dmarro89/dare-db/discussions/26

**Comment by dmarro89** [comment]
Closing the discussion, all the stuff about the documentation will continue here -> https://github.com/dmarro89/dare-db/discussions/26
